backend/plonk/bn254/solidity.go

```diff
@@ -85,14 +81,23 @@ library Utils {
 
     // reduce xmsg mod r, where xmsg is intrepreted in big endian
     // (as SetBytes does for golang's Big.Int library).
+    // ****************************************************
+    // NB: IF  res += uint256(xmsg[47 - i]) << (8 * i); will never overflow, move the unchecked around the for loop
```

Comment:
if this is moved around the `for loop`, the unchecked could be placed to span larger segments of code.